### ISSUE TITLE
Unable to build on android. IOS is working fine.

### ISSUE THREAD

**Issue by application-inoviotech** [stripe/stripe-react-native#1320]
**React Native Version : 0.68.2
Stripe Version : 0.25

Gradle version used : classpath('com.android.tools.build:gradle:7.0.2')**

* What went wrong:
Execution failed for task ':app:mergeDebugResources'.
> /Users/mac/.gradle/caches/transforms-3/b252794f6e7766bcaaa62afaf00adf95/transformed/material-1.8.0/res/values/values.xml: Error: Can't determine type for tag '<macro name="m3_comp_assist_chip_container_shape">?attr/shapeAppearanceCornerSmall</macro>'

**To Reproduce**
1) npx react-native init <projectname> --version 0.68.2
2) npm install @stripe/stripe-react-native
3) npm run android

**Expected behavior**
The project should work properly but it gives the above error

**Screenshots**
<img width="1680" alt="Screenshot 2023-03-01 at 12 38 10 PM" src="https://user-images.githubusercontent.com/109659590/222074424-35e7b335-045a-4310-8093-a77520f37618.png">

**Additional context**

When I use Gradle version "classpath('com.android.tools.build:gradle:7.0.2')". It gives me the above error on debug mode but when I upgrade my gradle version to "classpath('com.android.tools.build:gradle:7.2.2')", it starts working on debug mode but on release mode it gives me an error "Failed to get React Application Context". 

The main issue, I think is releated to assets merging because of the file react.gradle in react-native library. Please assist if someone has faced this issue or knows a workaround for it.

Currently I am using **Stripe Version 0.19** and it is working fine but there are alot of fixes done after this version so I wanted to upgrade my library.

**Comment by DhirenAtodaria** [comment]
I'm facing this same issue, it has to do with com.google.android.material:material:1.8.0 not working correctly with gradle tools below 7.2.2

If you aren't using payment sheets then an easy fix is to force the version to be 1.6.0 like so in your app's dependancy block:
```
configurations.all {
    resolutionStrategy {
        force 'com.google.android.material:material:1.6.0'
    }
}
```

**Comment by charliecruzan-stripe** [comment]
> 7.2.2, which itself requires react native 0.69+

@DhirenAtodaria does RN have that requirement? afaik, if your actual gradle version is 7.3.3 (check your `gradle-wrapper.properties`), you should be able to upgrade to `classpath("com.android.tools.build:gradle:7.2.2")` in your `build.gradle` file

**Comment by application-inoviotech** [comment]
@charliecruzan-stripe yes we're able to upgrade to **classpath("com.android.tools.build:gradle:7.2.2")** but that doesn't find react application context on release build. I don't know why is that but the issue is with merging assets. It says make sure your app is bundled correctly.

**Comment by DhirenAtodaria** [comment]
> > 7.2.2, which itself requires react native 0.69+
> 
> @DhirenAtodaria does RN have that requirement? afaik, if your actual gradle version is 7.3.3 (check your `gradle-wrapper.properties`), you should be able to upgrade to `classpath("com.android.tools.build:gradle:7.2.2")` in your `build.gradle` file

Yeah sorry, requirement is the wrong word, I'd say 'bugged' is more suitable 😅, but yeah like the poster above, we can upgradle to gradle tools 7.2.2, the problem occurs when you build a release version of the app, apparently there's some issues with the actual bundling of the js and it ends up causing the 'FATAL EXCEPTION: create_react_context' you can read more here: https://stackoverflow.com/questions/72564710/react-native-crash-release-build-fatal-exception-create-react-context

There were some suggested fixes but I deemed it wasn't worth the effort figuring it out and debugging after every release build for a simple library upgrade so ended up just forcing the material version since I don't use payment sheets regardless.

**Comment by application-inoviotech** [comment]
@DhirenAtodaria the issue is with some configuration in react.gradle file present in the react-native library itself. I tried upgrading the react native then everything works fine but I cannot upgrade react native version of my app.

**Comment by BhavikWings** [comment]
Got the same issue in "com.android.tools.build:gradle:7.2.1" , "com.google.android.material:material:1.6.0"

**Comment by echolove38** [comment]
> I'm facing this same issue, it has to do with com.google.android.material:material:1.8.0 not working correctly with gradle tools below 7.2.2, which itself requires react native 0.69+
> 
> If you aren't using payment sheets then an easy fix is to force the version to be 1.6.0 like so in your app's dependancy block:
> 
> ```
> configurations.all {
>     resolutionStrategy {
>         force 'com.google.android.material:material:1.6.0'
>     }
> }
> ```

where to add this ? which file ?

**Comment by DhirenAtodaria** [comment]
> > I'm facing this same issue, it has to do with com.google.android.material:material:1.8.0 not working correctly with gradle tools below 7.2.2, which itself requires react native 0.69+
> > If you aren't using payment sheets then an easy fix is to force the version to be 1.6.0 like so in your app's dependancy block:
> > ```
> > configurations.all {
> >     resolutionStrategy {
> >         force 'com.google.android.material:material:1.6.0'
> >     }
> > }
> > ```
> 
> where to add this ? which file ?

Hey, sorry for the late reply, but this goes in your build.gradle in your dependencies block

**Comment by arham1999** [comment]
I resolve this issue by using following configuration ,

`android/build.gradle`

```
dependencies {
...
classpath("com.android.tools.build:gradle:7.2.2")
classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:1.8.10")
...
}
```

`android/app/build.gradle`

```
dependencies {
...
implementation 'com.google.android.material:material:1.6.0'
...
}
```

NOTE: distribution url in gradle-wrapper.properties must not be lower than `distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-all.zip`

**Comment by MuhammadHussain-code** [comment]
> I resolve this issue by using following configuration ,
> 
> `android/build.gradle`
> 
> ```
> dependencies {
> ...
> classpath("com.android.tools.build:gradle:7.2.2")
> classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:1.8.10")
> ...
> }
> ```
> 
> `android/app/build.gradle`
> 
> ```
> dependencies {
> ...
> implementation 'com.google.android.material:material:1.6.0'
> ...
> }
> ```
> 
> NOTE: distribution url in gradle-wrapper.properties must not be lower than `distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.3-all.zip`

This solved my issue Thank you

**Comment by porter-stripe** [comment]
@application-inoviotech  We haven't heard from you in a bit, so we're going to close this issue. Please reopen if needed!
